### PR TITLE
[DCOS-2379] Print config changes to stdout in `dcos config set` and `dcos config …

### PR DESCRIPTION
**Pull request by mgummelt**
…unset`

### CODE REVIEW

File: cli/dcoscli/config/main.py [hunk 1]
```diff
@@ -170,6 +174,17 @@ def _set(name, value):
     _check_config(toml_config_pre, toml_config)
 
     config.save(toml_config)
+
+    if not value_exists:
+        emitter.publish("[{}] set to '{}'".format(name, python_value))
+    elif old_value == python_value:
```
Comment (ssk2) [inline cli/dcoscli/config/main.py]: Can we change python_value to new_value so it's obvious what's happening? I was a little confused ate first by "old_value" and "python_value".

File: cli/dcoscli/config/main.py [hunk 2]
```diff
@@ -237,19 +252,31 @@ def _unset(name, index):
...
+                        name))
             if index < 0 or index >= len(value):
                 raise DCOSException(
                     'Index ({}) is out of bounds - possible values are '
                     'between {} and {}'.format(index, 0, len(value) - 1))
 
-            value.pop(index)
+            popped_value = value.pop(index)
+            emitter.publish(
+                "[{}]: removed element '{}' at index '{}'".format(
```
Comment (ssk2) [inline cli/dcoscli/config/main.py]: We might want to add another case lists of length 1 - this output is a little odd:
```➜  cli git:(DCOS-2379-show-config-changes) ✗ dcos config unset --index=2 package.sources
Index (2) is out of bounds - possible values are between 0 and 0```

Comment (ssk2) [inline cli/dcoscli/config/main.py]: The colon after the property name here is inconsistent with output from other commands. 

E.g.:
```
➜  cli git:(DCOS-2379-show-config-changes) ✗ dcos config set core.timeout 5
[core.timeout] changed from '3' to '5'
```

And:
```
➜  cli git:(DCOS-2379-show-config-changes) ✗ dcos config unset --index=1 package.sources
[package.sources]: removed element 'https://github.com/mesosphere/universe/archive/version-1.y.zip' at index '1'
```

Is the name at the front a strict requirement (which isn't a bad idea)? If so, add a colon to the other messages. Alternatively, change this to `"Removed element '{}' at index '{}' from [{}]"`

File: cli/dcoscli/config/main.py [hunk 3]
```diff
@@ -237,19 +252,31 @@ def _unset(name, index):
         if index is not None:
             index = util.parse_int(index)
 
+            if not value:
+                raise DCOSException(
+                    'Index ({}) is out of bounds - {} is empty'.format(
```
Comment (ssk2) [inline cli/dcoscli/config/main.py]: No square brackets around package name in this message.

**Comment by ssk2** [comment]
Nice usability work @mgummelt! Thanks!

File: cli/dcoscli/config/main.py (hunk 1 above)
Comment (mgummelt) [inline cli/dcoscli/config/main.py]: done

File: cli/dcoscli/config/main.py (hunk 2 above)
Comment (mgummelt) [inline cli/dcoscli/config/main.py]: added colons

File: cli/dcoscli/config/main.py (hunk 3 above)
Comment (mgummelt) [inline cli/dcoscli/config/main.py]: fixed

**Comment by ssk2** [comment]
Great, merge when green please @mgummelt !

**Comment by tamarrow** [comment]
@mgummelt can we delete this branch?

**Pull request merged**